src/contents/button.tsx

```diff
@@ -24,12 +26,26 @@ class ButtonContributionManager {
   }
 
   _disabled = false;
+  _storage = new Storage();
 
   constructor(private contributions: ButtonContributionParams[]) {
     if (!this._disabled) {
       const isSuitable = isSiteSuitable();
       if (!isSuitable) {
         this._disabled = true;
+        if (isSiteGitpod()) {
+          sessionStorage.setItem("browser-extension-installed", "true");
+          (async () => {
+            const gitpodEndpoint = await this._storage.getItem(STORAGE_KEY_ADDRESS);
+            const host = new URL(gitpodEndpoint).host;
+            const thisPageHost = window.location.host;
+            if (host === thisPageHost) {
+              sessionStorage.setItem("browser-extension-active", "true");
```

Comment:
We use sessionStorage instead of localStorage, because we want it to be removed on tab close. This is the only way to detect that the browser extension is no longer installed.

I also thought about having the extension hide an element instead, but decided against it because it could lead to flashing the element for a frame or two.